### PR TITLE
a-tiny-js-world-OOP

### PR DESCRIPTION
**Pull request by NadiaVorontsova**
### A-tiny-js-world-OOP

[Demo](https://nadiavorontsova.github.io/a-tiny-JS-world-OOP/) |
[Code base](https://github.com/NadiaVorontsova/a-tiny-JS-world-OOP)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by NadiaVorontsova** [comment]
I have finished self-checks

File: submissions/NadiaVorontsova/a-tiny-JS-world-OOP/index.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+const outputElement = document.createElement('div');
+const mainBlock = document.querySelector('main');
+mainBlock.appendChild(outputElement);
+
+const dog = new Dog('Ben', 'male');
+const cat = new Cat('Sarah', 'female', 'Halle Berry');
+const man = new Man('Yura', 'male', 'Oksana');
+const woman = new Woman('Oksana', 'female', 'Yura');
+const catWoman = new CatWoman('Halle Berry', 'female', 'Sarah');
```
Comment (OleksiyRudenko) [inline submissions/NadiaVorontsova/a-tiny-JS-world-OOP/index.js:85]: WHy do we want users to specify gender when creating men and women?

File: submissions/NadiaVorontsova/a-tiny-JS-world-OOP/index.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+   constructor(species, name, gender, saying, friend) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+      this.friend = friend || 'no friend:(';
+   }
+   printInhabitant() {
+      return `I am a <strong>${this.species}</strong>; My name is <strong>${this.name}</strong>; I am a <strong>${this.gender}</strong>;
+      I say: <strong>'${this.saying}'</strong>; My friends: <strong>${this.friend}</strong>`;
```
Comment (OleksiyRudenko) [inline submissions/NadiaVorontsova/a-tiny-JS-world-OOP/index.js:18]: Not required to implement here. Just an idea for one of possible ways to add readability for long strings.

```
[
  "text",
  this.property,
  "more text",
  this.anotherProperty,
].join("; ");
```

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@NadiaVorontsova great job!
Let's take a bit of care of our peer developers that may be tasked to add 100's of men and women.

File: submissions/NadiaVorontsova/a-tiny-JS-world-OOP/index.js (hunk 1 above)
Comment (NadiaVorontsova) [reply submissions/NadiaVorontsova/a-tiny-JS-world-OOP/index.js]: oops it was by accident. I changed it!

File: submissions/NadiaVorontsova/a-tiny-JS-world-OOP/index.js (hunk 2 above)
Comment (NadiaVorontsova) [reply submissions/NadiaVorontsova/a-tiny-JS-world-OOP/index.js]: Thanks for this example. I used it)

File: submissions/NadiaVorontsova/a-tiny-JS-world-OOP/index.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+   }
+}
+
+class Human extends Inhabitant {
+   constructor(name, gender, saying, friend) {
+      super('human', name, gender, saying, friend);
+      this.legs = 2;
+      this.hands = 2;
+   }
+   printHuman() {
```
Comment (OleksiyRudenko) [inline submissions/NadiaVorontsova/a-tiny-JS-world-OOP/index.js:33]: What if we could rename this method as `printInhabitant`? Same refers to `printAnimal`.
We probably would not need separate lists for people and animals at the end of the code.

There is a problem -- the method will try to call itself in an endless loop under current implementation due to **this**.printInhabitant.

See how [`super`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/super) can help you out.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@NadiaVorontsova well done.
There is yet some room for improvement and discovery.

File: submissions/NadiaVorontsova/a-tiny-JS-world-OOP/index.js (hunk 3 above)
Comment (NadiaVorontsova) [reply submissions/NadiaVorontsova/a-tiny-JS-world-OOP/index.js]: I did it) cool

**Review by OleksiyRudenko — APPROVED** [review]
@NadiaVorontsova great job done!

**Comment by NadiaVorontsova** [comment]
> @NadiaVorontsova great job done!

Thank you)

**Pull request merged**